### PR TITLE
EYZ-439 Batch Update records

### PR DESCRIPTION
**Pull request by dseeker**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -733,28 +733,52 @@ class Zoho {
...
+            )
+        }
+
+        let resultData = { records: [], hasMore: true };
+
+        try {
+            let results = []
+            while (allPromises.length){
+                if (module_options.debug) console.log('ZohoAPI updateRecords put', allPromises.length, module_options.records_batch_size);
+                results.push( await Promise.all(allPromises.splice(0, module_options.records_batch_size)) )
```
[the remark this answers is not in the document]
Comment (dseeker) [reply index.js]: no, per page is the number o items in each data object, this is the concurrency of calls

**Comment by dseeker** [comment]
> 
> 
> needs your input. does not run with the following code
> 
> ```
> try {
>         let module = "Catalogue";
>         let result = await zoho.updateRecords(module, data);
>         console.log(result);
>     } catch (error) {
>         console.log(error);
>     }
> ```
> 
> breaks at line with flat. please also other comment

Flat was introduced in NODE 11, try upgrading

File: index.js [hunk 2]
```diff
@@ -733,28 +733,52 @@ class Zoho {
...
+            }
+            results = results.flat()
```
[the remark this answers is not in the document]
Comment (dseeker) [reply index.js]: what in the data object? what's the error?

**Comment by phanisaripalli** [comment]
data

```
let data = [
        {id: "1972094000018824820", trivia: false},
        {id: "1972094000021014015", trivia: false},
    ];
```

**Comment by phanisaripalli** [comment]
But I guess it is running because I am on 10. Dont think I am ready to upgrade.
Can this be done without flat.

**Comment by dseeker** [comment]
I couldn't find an easier way, give it a try if you want

**Comment by phanisaripalli** [comment]
@dseeker could you please check

**Pull request merged**